### DISCUSSION TITLE
trusted Proxy - OpnSense HA Proxy - get real client IP

### DISCUSSION THREAD

**Discussion by v1net** [dani-garcia/vaultwarden#4904] [Q&A]
hey iam using OpnSense with HA Proxy

with my nextcloud i need to set the "trusted proxy" Ips to see in my log files the public wan ip from the client who wants to connect or tryed to login. 

on vaultwarden lxc container i didnt finde a solution how i can see the public wan ip in the logs. In the HA Proxy i had the same settings as for the vaultwarden. 

Maybe someone else has here a haproxy and a vaultwarden behind?

backend VaultWarden
    # health checking is DISABLED
    mode http
    balance source
    # stickiness
    stick-table type ip size 50k expire 30m  
    stick on src
    http-reuse safe
    option forwarded for
    option forwardfor
    server VaultWarden X.X.X.X:443 ssl alpn http/1.1 verify none

frontend 443_public
    http-response set-header Strict-Transport-Security "max-age=15768000; includeSubDomains; preload"
    bind 0.0.0.0:443 name 0.0.0.0:443 ssl prefer-client-ciphers ssl-min-ver TLSv1.2 ciphers ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256 ciphersuites TLS_AES_128_GCM_SHA256:TLS_AES_256_GCM_SHA384:TLS_CHACHA20_POLY1305_SHA256 no-alpn crt-list /tmp/haproxy/ssl/64381b3ed51447.31363174.certlist 
    mode http
    option http-keep-alive
    option forwardfor

# Frontend: 80_public_redirect_443 ()
frontend 80_public_redirect_443
    bind 0.0.0.0:80 name 0.0.0.0:80 
    mode http
    option http-keep-alive
    option forwardfor

    # logging options
    # ACL: http
    acl acl_64381ca0891a22.74632366 req.proto_http

    # ACTION: http-to-https
    http-request redirect scheme https if acl_64381ca0891a22.74632366

**Comment by BlackDex** [comment]
As mentioned on Matrix also.
Check the proxy examples, https://github.com/dani-garcia/vaultwarden/wiki/Proxy-examples.

There you see:
```haproxy
backend vaultwarden_http
    # ~CUT SOME COMMENTS~
    # Add the x-forwarded-for header
    option forwardfor
    # Set the Source IP in the `X-Real-IP` header
    http-request set-header X-Real-IP %[src]
    # Send the traffic to the local instance
    server vwhttp 0.0.0.0:8080 alpn http/1.1
```

Basically just add `http-request set-header X-Real-IP %[src]` to your backend.
Make sure Vaultwarden has either the default (unset) `IP_HEADER` or set it to `X-Real-IP`.

**Comment by v1net** [comment]
this solved my problem

add this lines in this config: 172.17.2.0 is my gateway network

        set_real_ip_from 172.17.2.0/24;
        real_ip_header X-Forwarded-For;



 /etc/nginx/nginx.conf                                                            
user www-data;
worker_processes auto;
pid /run/nginx.pid;
error_log /var/log/nginx/error.log;
include /etc/nginx/modules-enabled/*.conf;

events {
        worker_connections 768;
        # multi_accept on;
}

http {

        ##
        # Basic Settings
        ##
        set_real_ip_from 172.17.2.0/24;
        real_ip_header X-Forwarded-For;

**Reply by BlackDex** [reply]
That is something totally different from HAProxy.
